tritium-registry/src/test/java/com/palantir/tritium/metrics/registry/MetricNameTest.java

```diff
@@ -36,11 +36,13 @@ public void compareSame() {
                 .putSafeTags("key", "value")
                 .putSafeTags("key1", "value1")
                 .build();
+        MetricName three = RealMetricName.create(two);
 
-        assertThat(one).isEqualTo(two);
-        assertThat(two).isEqualTo(one);
+        assertThat(one).isEqualTo(two).isEqualTo(three);
+        assertThat(two).isEqualTo(one).isEqualTo(three);
+        assertThat(three).isEqualTo(one).isEqualTo(two);
```

Comment:
Does assertj have any equivalent to perform the full equals/hashCode comparison reflexive, symmetric, transitive permutations like `com.google.common.testing.EqualsTester` [`org.assertj.core.test.EqualsHashCodeContractAssert`](https://github.com/joel-costigliola/assertj-core/blob/master/src/test/java/org/assertj/core/test/EqualsHashCodeContractAssert.java) seems to be what we'd want, but not sure why its not in main assertj module.

Might also want to verify `hashCode` while we're at it.
```
        MetricName four = ImmutableMetricName.copyOf(three);

        assertThat(one)
                .hasSameHashCodeAs(two)
                .hasSameHashCodeAs(three)
                .hasSameHashCodeAs(four);
```

Reply:
> Does assertj have any equivalent to perform the full equals/hashCode comparison reflexive, symmetric, transitive permutations

Not that I'm aware of, I can file an assertj FR upstream though.

> Might also want to verify hashCode while we're at it.

Good idea, done.